### ISSUE TITLE
Extend RemoveDir to allow removing of directories

### ISSUE THREAD

**Issue by Leandros** [fastbuild/fastbuild#147]
Currently, as far as I understand, `RemoveDir` currently only support removing the _contents_ of a directory, not the directory itself.

Would be great if this could be extended to allow removal of the directory itself.

**Comment by ffulin** [comment]
Seems like we could add a `.RemoveDirs` option (that defaults to false) which could remove the directory/directories after deleting the files inside.

**Comment by Leandros** [comment]
Sounds like a good approach!

**Comment by ffulin** [comment]
RemoveDir has been extended to remove directories as well as the files within them: https://github.com/fastbuild/fastbuild/commit/4c547b460c37f7cd8d7fcd7918c6cb2e3be35e9c

I ended up making this the default behavior, as even though it's a change in behavior, I think it's generally the expected behavior. I added additional options to allow removal of directories to be disabled.

This change will be in the next release (v1.12). I'll leave this issue open until that version is released.
